src/aws.rs

```diff
@@ -0,0 +1,196 @@
+        }
+
+        let aws_users = self.get_credential_report();
+
+        return Ok(GetUsersResult{
+                    service_name: "aws".to_string(),
+                    users: aws_users.iter().filter_map(|user|
+                        match user.mfa_active {
+                            false => Some(User{
+                                            name: user.user.to_string(),
+                                            email: None,
+                                            details: Some(user.arn.to_string())
+                                          }),
+                            _ => None
+                        }).collect()
+                    });
+    }
+}
+
+#[derive(Debug)]
+enum ReportResult {
+    STARTED,
+    COMPLETE,
+}
+
+impl ReportResult {
+    fn from_str(s: &str) -> ReportResult {
+        match s {
+            "STARTED" => ReportResult::STARTED,
+            "COMPLETE" => ReportResult::COMPLETE,
+            _ => panic!("Unknown report result.")
+        }
+    }
+}
+
+#[derive(RustcDecodable,Debug)]
+struct AWSUser {
+    user: String,
+    arn: String,
+    user_creation_time: String,
+    password_enabled: String,
+    password_last_used: String,
+    password_last_changed: String,
+    password_next_rotation: String,
+    mfa_active: bool,
+    access_key_1_active: String,
+    access_key_1_last_rotated: String,
+    access_key_1_last_used_date: String,
+    access_key_1_last_used_region: String,
+    access_key_1_last_used_service: String,
+    access_key_2_active: String,
+    access_key_2_last_rotated: String,
+    access_key_2_last_used_date: String,
+    access_key_2_last_used_region: String,
+    access_key_2_last_used_service: String,
+    cert_1_active: String,
+    cert_1_last_rotated: String,
+    cert_2_active: String,
+    cert_2_last_rotated: String,
+}
+
+impl AWSService {
+    fn request(&self, action: Action) -> Response {
+        let region = Region::UsEast1;
+        let mut request = SignedRequest::new(
+            "GET",
+            "iam",
+            &region,
+            "/",
+        );
+
+        request.add_param("Version", "2010-05-08");
+        request.add_param("Action", &action.as_str());
+
+        let credentials = AWSCredentials::new(
+            self.access_key_id.as_ref(),
+            self.secret_key.as_ref(),
+            None,
+            UTC::now() + Duration::seconds(600));
+
+        return request.sign_and_execute(&credentials)
+    }
+
+    fn generate_credential_report(&self) -> ReportResult {
+        let response = self.request(Action::GenerateCredentialReport);
+        let mut tree = Element::parse(response);
+        let state = tree.get_mut_child("GenerateCredentialReportResult")
+                        .expect("Can't find GenerateCredentialReportResult element.")
+                        .get_mut_child("State")
+                        .expect("Can't find State element.");
+
+        return match state.text {
```

Comment:
`unwrap()` moves but the state struct is borrowed, so just using unwrap results in the following a "cannot move from borrowed content".

See also http://is.gd/AQXPQR.

It's possible that there is a different solution that would work, but I didn't find it.